File: Lecture_Example/week5/self_demo.py
import math

class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def size(self):
        return math.sqrt(self.x ** 2 + self.y ** 2)

    def same_radius(self, other):
        return self.size() == other.size()

File: Lecture_Example/week5/self_demo.js
class Point {
    constructor(x, y) {
        this.x = x;
        this.y = y;
    }

    size() {
        return Math.sqrt(this.x * this.x + this.y * this.y);
    }

    sameSize(other) {
        return this.size() === other.size();
    }
}
